operator_ui/src/pages/Jobs/utils.test.tsx

```diff
@@ -135,6 +139,20 @@ describe('pages/jobs/utils', () => {
       ).toEqual({
         format: JobSpecFormats.TOML,
         list: false,
+        error: '',
+      })
+    })
+
+    it('returns an error on duplicate parents', () => {
+      expect(
+        getTaskList({
+          value:
+            'observationSource = """ ds -> ds_parse; ds1 -> ds1_parse; ds -> ds_parse;  """',
+        }),
+      ).toEqual({
+        format: JobSpecFormats.TOML,
+        list: false,
+        error: 'ds has duplicate ds_parse children',
```

Comment:
I am surprised that this data is allowed to be in the database at all if it's invalid DOT. This feels like a backend bug to me, do you have an example that causes this?

Reply:
This is the example from the ticket. Prior to the fix it would crash the UI, but after fixing the crash and attempting to submit it would get created. I believe the validator only checks if it's a valid toml.
```
name = "dYdX 1INCH / USD -- STAGING"
type = "cron"
schemaVersion = 1
schedule = "* * * * *"
observationSource = """
   // data source 1
   ds1          [type=bridge name="bridge-coinapi" requestData="{\\"data\\": {\\"base\\":\\"1INCH\\",\\"quote\\":\\"USD\\"}}"];
   ds1_parse    [type=jsonparse path="result"];
   ds1_multiply [type=multiply times=100000000];
   // data source 2
   ds2          [type=bridge name="bridge-coingecko" requestData="{\\"data\\": {\\"base\\":\\"1INCH\\",\\"quote\\":\\"USD\\"}}"];
   ds2_parse    [type=jsonparse path="result"];
   ds2_multiply [type=multiply times=100000000];
   // data source 1
   ds3          [type=bridge name="bridge-nomics" requestData="{\\"data\\": {\\"base\\":\\"1INCH\\",\\"quote\\":\\"USD\\"}}"];
   ds3_parse    [type=jsonparse path="result"];
   ds3_multiply [type=multiply times=100000000];
   medianized_answer [type=median];
   submit            [type=bridge name="bridge-dydx-stark-prod" includeInputAtKey="result" requestData="{\\"data\\": {\\"asset\\":\\"1INCHUSD\\"}}"]
   // pipeline graph
   ds1 -> ds1_parse -> ds1_multiply -> medianized_answer;
   ds2 -> ds2_parse -> ds2_multiply -> medianized_answer;
   ds2 -> ds2_parse -> ds2_multiply -> medianized_answer;
   medianized_answer -> submit;
"""
```

Reply:
Regarding the bug, it's both on the frontend and backend. In the meantime this ticket changes fix the UI crash on input and prevent submittal of jobs with duplicates from the UI to the backend.

Reply:
Can we fix it on backend as well and include a test?

Reply:
@samsondav 
The backend fix is outside of the scope of this ticket.
There is a separate ticket for that.